hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -366,6 +374,14 @@ connectionFactory, getOzoneManagerSnapshotUrl(),
       reconUtils.untarCheckpointFile(targetFile, untarredDbDir);
       FileUtils.deleteQuietly(targetFile);
 
+      // Validate the presence of required SST files
+      File[] sstFiles = untarredDbDir.toFile().listFiles((dir, name) -> name.endsWith(".sst"));
+      if (sstFiles == null || sstFiles.length == 0) {
+        LOG.warn("No SST files found in the OM snapshot directory: {}", untarredDbDir);
+        return null;
+      }
+      LOG.info("Valid OM snapshot with SST files found at: {}", untarredDbDir);
```

Comment:
In addition to debug level, I would request to log the list of valid SST files, so in case of issues where compaction of SST files happens at OM, SST files fetched may not be exactly matching at Recon side, or another scenario where due to some unexpected issue at OM, SST files being fetched but not all being fetched since the last sequence number, then printing the list of SST files along with corresponding sequence number as debug log would be helpful to debug.

1. Add the last sequence number somewhere in DB under ReconTask table with the corresponding task details and list of SST files, if not done yet. This can be added later in troubleshooting doc in case of OM DB display data and customer can get the details from table.
2. Same thing need to do for other tasks like NSSummary tasks which may impact DU and customer should be able to troubleshoot on its own.